src/GloryDuckTables2019Lkl.cc

```diff
@@ -0,0 +1,403 @@
+
+  // get first line of the file (with <sv> values)
+  string sigmavLine;
+  getline(ff,sigmavLine);
+
+  // read first line of the file
+  std::istringstream row(sigmavLine);
+  Double_t field;
+  while (row>> field)
+    {
+      vsigma.push_back(field);
+    }
+
+  Double_t readingMass = 0.;
+  Double_t readingLkl = 0.;
+  Int_t counter = 0;
+
+  // get mass (first column)
+  while(ff >> readingMass)
+    {
+      // get rest of the line (<sv> values)
+      string LklLine;
+      getline(ff,LklLine);
+
+      // read rest of the line
+      std::istringstream line(LklLine);
+      while (line >> readingLkl)
+        {
+          vlkl.push_back(readingLkl);
+        }
+
+      // add new parabola
+      CreateAndAddNewParabola(readingMass,vsigma[0],vsigma[vsigma.size()-1],vsigma.size(),vsigma.data(),vlkl.data());
+
+      // check if line was complete or not
+      if(vlkl.size() != vsigma.size())
+        {
+          cout << "You have a different line size length for line " << counter+1 << " ! Number of sigmav values = " << vsigma.size() << " while number of likelihood values = " << vlkl.size() << endl; 
+        }
+      counter++;
+      vlkl.clear();
+   }
+
+  ff.close(); 
+
+  SetChecked(kFALSE);
+
+  cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Message: Read " <<  GetSampleArray()->GetEntries() << " parabolas for a total of " << fNMasses << " DM masses." << endl;
+  return 0;
+}
+
+////////////////////////////////////////////////////////////////
+//
+// Add a new ParabolaLkl object representing the -2logL values vs
+// <sv> for one DM mass
+//
+// Return 0 in case of success, 1 otherwise
+//
+Int_t GloryDuckTables2019Lkl::CreateAndAddNewParabola(Double_t mass,Double_t svmin,Double_t svmax,Int_t npoints,Double_t* sv,Double_t* logL) 
+{
+  fNMasses++;
+
+  // Add new bin bounds to the list
+  Double_t* newMassList = new Double_t[fNMasses];
+  Double_t* newSvMinList = new Double_t[fNMasses];
+  Double_t* newSvMaxList = new Double_t[fNMasses];
+
+  // copy old lists into new ones
+  if(fNMasses>1)
+    {
+      for(Int_t ibin=0;ibin<fNMasses-1;ibin++)
+	{
+	  newMassList[ibin] = fMass[ibin];
+	  newSvMinList[ibin] = fsvMin[ibin];
+	  newSvMaxList[ibin] = fsvMax[ibin];
+	}
+      delete [] fMass;
+      delete [] fsvMin;
+      delete [] fsvMax;
+    }
+
+  newMassList[fNMasses-1] = mass;
+  newSvMinList[fNMasses-1] = svmin;
+  newSvMaxList[fNMasses-1] = svmax;
+  fMass = newMassList;
+  fsvMin = newSvMinList;
+  fsvMax = newSvMaxList;
+
+  // Create and add the new ParabolaLkl object
+  ParabolaLkl* newparabola = new ParabolaLkl(npoints,sv,logL,Form("ParabolaLkl_%d",GetSampleArray()->GetEntries()));
+  AddSample(newparabola);
+  
+  SetChecked(kFALSE);
+  return 0;
+}
+
+//////////////////////////////////////////////////////////////////
+//
+// Function called by Lkl::PrintOverview
+// Print info about the experimental data
+//
+void GloryDuckTables2019Lkl::PrintData(Int_t level)
```

Comment:
Print also all parabolas contents. Maybe later can be removed if we get too messy logs, but for test purposes it will be useful